docs/src/model_implementation.md

```diff
@@ -0,0 +1,327 @@
+                                     required_biogeochemical_auxiliary_fields,
+                                     update_biogeochemical_state!,
+                                     biogeochemical_drift_velocity,
+				                     biogeochemical_auxiliary_fields
+```
+
+We then define our `struct` with the model parameters, as well as slots for the particles, light attenuation, and sediment models:
+
+```@example implementing
+@kwdef struct NutrientPhytoplankton{FT, LA, S, W, P} <:ContinuousFormBiogeochemistry{LA, S, P}
+            base_growth_rate :: FT = 1.27 / day              # 1 / seconds
+    nutrient_half_saturation :: FT = 0.025 * 1000 / 14       # mmol N / m³
+       light_half_saturation :: FT = 300.0                   # micro einstein / m² / s
+        temperature_exponent :: FT = 0.24                    # 1
+     temperature_coefficient :: FT = 1.57                    # 1
+         optimal_temperature :: FT = 28.0                    # °C
+              mortality_rate :: FT = 0.15 / day              # 1 / seconds
+     crowding_mortality_rate :: FT = 0.004 / day / 1000 * 14 # 1 / seconds / mmol N / m³
+
+     light_attenuation_model :: LA = nothing
+              sediment_model :: S  = nothing
+            sinking_velocity :: W  = 2 / day
+                   particles :: P  = nothing
+end
+```
+
+In the above we used `@kwdef` to set default values for the models so that we don't have to set these up each time we use the model. We have also included a `sinking_velocity` field to later demonstrate how we can get tracers to sink. We also need to define some functions so that OceanBioME and Oceananigans know what tracers and auxiliary fields (e.g. light intensity) we need setting up:
+
+```@example implementing
+required_biogeochemical_tracers(::NutrientPhytoplankton) = (:N, :P, :T)
+
+required_biogeochemical_auxiliary_fields(::NutrientPhytoplankton) = (:PAR, )
+
+biogeochemical_auxiliary_fields(bgc::NutrientPhytoplankton) = biogeochemical_auxiliary_fields(bgc.light_attenuation_model)
+```
+
+Next we need to define the methods that specify how the nutrient, ``N``, and phytoplankton ``P`` evolve. We want the phytoplankton to evolve at the rate given by:
+
+```math
+\frac{\partial P}{\partial t} = \mu g(T) f(N) h(PAR) P - mP - bP^2,
+```
+
+where ``\mu`` is `base_growth_rate`, ``m`` is `mortality_rate`, ``b`` is `crowding`, and ``g``, ``f``, and ``h`` are given by:
+```math
+\begin{align}
+g(T) &= c_1\exp\left(-c_2|T - T_{opt}|\right),\\
+f(N) &= \frac{N}{k_N + N},\\
+h(PAR) &= \frac{PAR}{k_P + PAR},
+\end{align}
+```
+where ``c_1`` and ``c_2`` are `temperature_coefficient` and `temperature_exponent`, ``T_{opt}`` is `optimal_temperature`, ``k_N`` is `nutrient_half_saturation`, and ``k_P`` is `light_half_saturation`. Since this is a simple two variable model where the total concentration is conserved, ``\frac{\partial N}{\partial t} = - \frac{\partial P}{\partial t}``.
```

Comment:
```suggestion
where ``c_1`` is `temperature_coefficient`,  ``c_2`` is `temperature_exponent`, ``T_{opt}`` is `optimal_temperature`, ``k_N`` is `nutrient_half_saturation`, and ``k_P`` is `light_half_saturation`. Since this is a simple two variable model and the total concentration is conserved, ``\frac{\partial N}{\partial t} = - \frac{\partial P}{\partial t}``.
```